internal/authentication/password_hash.go

```diff
@@ -149,12 +134,57 @@ func CheckPassword(password, hash string) (ok bool, err error) {
 }
 
 func getCryptSettings(salt string, algorithm CryptAlgo, iterations, memory, parallelism, keyLength int) (settings string) {
-	if algorithm == HashingAlgorithmArgon2id {
+	switch algorithm {
+	case HashingAlgorithmArgon2id:
 		settings, _ = crypt.Argon2idSettings(memory, iterations, parallelism, keyLength, salt)
-	} else if algorithm == HashingAlgorithmSHA512 {
+
+	case HashingAlgorithmSHA512:
 		settings = fmt.Sprintf("$6$rounds=%d$%s", iterations, salt)
-	} else {
+	default:
 		panic("invalid password hashing algorithm provided")
 	}
 	return settings
 }
+
+// validateSalt checks the salt input and settings are valid and returns it and a nil error if they are, otherwise returns an error.
+func validateSalt(salt string, saltLength int) (string, error) {
+	if salt == "" {
+		if saltLength < 2 {
+			return "", fmt.Errorf("Salt length input of %d is invalid, it must be 2 or higher", saltLength)
+		} else if saltLength > 16 {
+			return "", fmt.Errorf("Salt length input of %d is invalid, it must be 16 or lower", saltLength)
+		}
+	} else if len(salt) > 16 {
+		return "", fmt.Errorf("Salt input of %s is invalid (%d characters), it must be 16 or fewer characters", salt, len(salt))
+	} else if len(salt) < 2 {
+		return "", fmt.Errorf("Salt input of %s is invalid (%d characters), it must be 2 or more characters", salt, len(salt))
+	} else if _, err := crypt.Base64Encoding.DecodeString(salt); err != nil {
+		return "", fmt.Errorf("Salt input of %s is invalid, only characters [a-zA-Z0-9+/] are valid for input", salt)
+	}
+	if salt == "" {
```

Comment:
The salt generation does not really help validating the salt, I'd rather keep the block in the calling function or rename the function from `validateSalt` into something else.